app/api/resources/user.py

```diff
@@ -509,3 +509,96 @@ def get(cls):
             return messages.USER_NOT_FOUND, HTTPStatus.NOT_FOUND
 
         return dashboard, HTTPStatus.OK
+
+@users_ns.route("user/forgot_password")
+class ForgotPassword(Resource):
+    @classmethod
+    @users_ns.doc("forgot password")
+    @users_ns.response(HTTPStatus.OK, f"{messages.PASSWORD_RESET_MAIL_MESSAGE}")
+    @users_ns.response(HTTPStatus.BAD_REQUEST, f"{messages.EMAIL_FIELD_IS_MISSING}\n{messages.EMAIL_INPUT_BY_USER_IS_INVALID}")
+    @users_ns.response(HTTPStatus.FORBIDDEN, f"{messages.USER_HAS_NOT_VERIFIED_EMAIL_BEFORE_LOGIN}")
+    @users_ns.response(HTTPStatus.NOT_FOUND, f"{messages.USER_IS_NOT_REGISTERED_IN_THE_SYSTEM}")
+    @users_ns.expect(forgot_password_change_request_data_model, validate=True)
+    def post(self):
+        """ Updates User's password by sending reset link to email
+ 
+        Args: 
+          email : Verified email of the user
+ 
+        Return:
+          token: a token link/reset link will be shared on verified mail of user
+        """
+ 
+        data = request.json
+ 
+        if not data["email"]:
+            return messages.EMAIL_FIELD_IS_MISSING, HTTPStatus.BAD_REQUEST
+ 
+        is_valid = validate_forgot_password_email(data)
+ 
+        if is_valid != {}:
```

Comment:
```suggestion
        if not is_email_valid(data["email"]):
```
That function you have created `validate_forgot_password_email()` has this sole purpose